### PR TITLE
create package

### PR DESCRIPTION
**Pull request by wleoncio**
This PR reformats the repository content as an R package.

# Installation

Until this PR is merged, one convenient way to check out the package is by running the following in R:

```r
remotes::install_github("wleoncio/SparseFunClust", "create-package", build_vignettes = TRUE)
```

# Accessing the functions and vignette

At the moment, no functions are exported, so they all need to be accessed by prefixing them with `SparseFunClust:::` (notice the triple colons). Exporting functions will be done on a later PR adjusting all the documentation.

To view the (placeholder) vignette: `browseVignettes("SparseFunClust")`.

# Summary of changes

- Added package structure
- Moved R scripts to R folder
- Reformatted toy example as a vignette
- Adjusted vignette to run
- Removed calls to source() and functions from .Rbuildignore
- Added installation and vignette instructions to README
- Removed invalid argument to sqrt()
- Resolved imported function calls

### CODE REVIEW

**Comment by wleoncio** [comment]
A couple of things I've wondered:

1. Question: I assumed the `pam()` functions used in `FKMSparseClustering()` and `GetOptimalClusters()` both come from the `cluster` package. Is that the case?
2. `GetOptimalW()` had a call to `sqrt()` that used an invalid argument, `na.rm = TRUE`, which generated a warning. I removed that second argument. Hope that's ok.

**Comment by wleoncio** [comment]
Hi Valeria,

Commit [54087ff](https://github.com/valeriavitelli/SparseFunClust/pull/1/commits/54087ffb848fe7a07012e184f9a363eb92afec85) applies the changes made on the latest version of master. 

GitHub is still pointing to conflicts in the R scripts, probably because it's still confused about files that were moved. To double-check, I ran the command below locally, comparing the content of your ("origin") master branch and my ("fork") create-package branch:

bilde

Notice how all *.R functions contain no changes. The files with only "+" are added files and the file with some actual changes is the toy example, which was reformatted from R to Rmd.

In other words, now it's safe to resolve the conflicts by accepting the changes in this PR.

Let me know if I can help with any next steps!

**Comment by wleoncio** [comment]
Hi Valeria,

Commit [54087ff](https://github.com/valeriavitelli/SparseFunClust/pull/1/commits/54087ffb848fe7a07012e184f9a363eb92afec85) applies the changes made on the latest version of master (https://github.com/valeriavitelli/SparseFunClust/pull/1/commits/54087ffb848fe7a07012e184f9a363eb92afec85). 

GitHub is still pointing to conflicts in the R scripts, probably because it's still confused about files that were moved. To double-check, I ran the command below locally, comparing the content of your ("origin") master branch and my ("fork") create-package branch:

bilde

Notice how all *.R functions contain no changes. The files with only "+" are added files and the file with some actual changes is the toy example, which was reformatted from R to Rmd.

In other words, now it's safe to resolve the conflicts by accepting the changes in this PR.

Let me know if I can help with any next steps!

**Review by valeriavitelli — APPROVED** [review]
